### PR TITLE
Dev 1.4 abstract classes and interfaces

### PR DESCRIPTION
**Pull request by romantanchinsky**

### CODE REVIEW

File: HomeWorks/DEV-1.4_AbstractClassesAndInterfaces/AbstractClassesAndInterfaces/Entitys/Coordinate.cs [hunk 1]
```diff
@@ -0,0 +1,63 @@
+﻿using System;
+
```
Comment (IgorKheidorov) [inline HomeWorks/DEV-1.4_AbstractClassesAndInterfaces/AbstractClassesAndInterfaces/Entitys/Coordinate.cs:2]: Entities, grammar

File: HomeWorks/DEV-1.4_AbstractClassesAndInterfaces/AbstractClassesAndInterfaces/Entitys/Coordinate.cs [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+            }
+        }
+
+        public Coordinate ( double theX, double theY, double theZ )
+        {
+            X = theX;
+            Y = theY;
+            Z = theZ;
+        }
+
```
Comment (IgorKheidorov) [inline HomeWorks/DEV-1.4_AbstractClassesAndInterfaces/AbstractClassesAndInterfaces/Entitys/Coordinate.cs:57]: No need in a separate method, only if some non trivial logics

File: HomeWorks/DEV-1.4_AbstractClassesAndInterfaces/AbstractClassesAndInterfaces/Entitys/Coordinate.cs [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+        private bool IsPositive ( double number )
+        {
```
Comment (IgorKheidorov) [inline HomeWorks/DEV-1.4_AbstractClassesAndInterfaces/AbstractClassesAndInterfaces/Entitys/Coordinate.cs:59]: number for coordinate can be 0

File: HomeWorks/DEV-1.4_AbstractClassesAndInterfaces/AbstractClassesAndInterfaces/Entitys/Coordinate.cs (hunk 1 above)
Comment (romantanchinsky) [reply HomeWorks/DEV-1.4_AbstractClassesAndInterfaces/AbstractClassesAndInterfaces/Entitys/Coordinate.cs:2]: Renamed

File: HomeWorks/DEV-1.4_AbstractClassesAndInterfaces/AbstractClassesAndInterfaces/Entitys/Coordinate.cs (hunk 3 above)
Comment (romantanchinsky) [reply HomeWorks/DEV-1.4_AbstractClassesAndInterfaces/AbstractClassesAndInterfaces/Entitys/Coordinate.cs:59]: changed

File: HomeWorks/DEV-1.4_AbstractClassesAndInterfaces/AbstractClassesAndInterfaces/Entitys/Coordinate.cs (hunk 2 above)
Comment (romantanchinsky) [reply HomeWorks/DEV-1.4_AbstractClassesAndInterfaces/AbstractClassesAndInterfaces/Entitys/Coordinate.cs:57]: It's condition for coordinates, maybe wrong name. I changed to IsPermissible

**Pull request merged**